File: sdk-dotnet/TollaraHeaders.cs
namespace Tollara;

/// <summary>Canonical Tollara HTTP header names.</summary>
public static class TollaraHeaders
{
    public const string Signature = "X-Tollara-Signature";
    public const string Timestamp = "X-Tollara-Timestamp";
    public const string UserId = "X-Tollara-User-ID";
    public const string ServiceProductId = "X-Tollara-Service-Product-ID";
    public const string Roles = "X-Tollara-Roles";
    public const string SubscriptionStatus = "X-Tollara-Subscription-Status";
    public const string BillingModel = "X-Tollara-Billing-Model";
    public const string MeasurementType = "X-Tollara-Measurement-Type";
    public const string UnitLabel = "X-Tollara-Unit-Label";

    /// <summary>v1/v2 only; use <see cref="ServiceProductId"/> for signing version 3.</summary>
    [Obsolete("v1/v2 only; use ServiceProductId for signing version 3.")]
    public const string Plan = "X-Tollara-Plan";

    /// <summary>v1 only; omitted from v2/v3 HMAC material.</summary>
    [Obsolete("v1 only; omitted from v2/v3 HMAC material.")]
    public const string QuotaRemaining = "X-Tollara-Quota-Remaining";

    /// <summary>v1/v2 only; use <see cref="SubscriptionStatus"/> for signing version 3.</summary>
    [Obsolete("v1/v2 only; use SubscriptionStatus for signing version 3.")]
    public const string SubscriptionActive = "X-Tollara-Subscription-Active";

    /// <summary>Gateway HMAC user-context schema: <c>3</c> = v3; <c>2</c> = v2 (leading <c>2</c>, no quota segment).</summary>
    public const string SigningVersion = "X-Tollara-Signing-Version";
}

File: sdk-python/src/tollara_service_sdk/tollara_headers.py
"""Canonical Tollara HTTP header names."""


class TollaraHeaders:
    SIGNATURE = "X-Tollara-Signature"
    TIMESTAMP = "X-Tollara-Timestamp"
    USER_ID = "X-Tollara-User-ID"
    SERVICE_PRODUCT_ID = "X-Tollara-Service-Product-ID"
    ROLES = "X-Tollara-Roles"
    SUBSCRIPTION_STATUS = "X-Tollara-Subscription-Status"
    BILLING_MODEL = "X-Tollara-Billing-Model"
    MEASUREMENT_TYPE = "X-Tollara-Measurement-Type"
    UNIT_LABEL = "X-Tollara-Unit-Label"

    # v1/v2 only; use SERVICE_PRODUCT_ID for signing version 3
    PLAN = "X-Tollara-Plan"
    # v1 only; omitted from v2/v3 HMAC material
    QUOTA_REMAINING = "X-Tollara-Quota-Remaining"
    # v1/v2 only; use SUBSCRIPTION_STATUS for signing version 3
    SUBSCRIPTION_ACTIVE = "X-Tollara-Subscription-Active"

    # Gateway HMAC user-context schema: 3 = v3; 2 = v2 (leading "2", no quota segment)
    SIGNING_VERSION = "X-Tollara-Signing-Version"
